DOCSP-45860-older-version-limitations (#522)
mongodb/docs-cluster-to-cluster-sync: source/includes/fact-older-version-limitations.rst

- Writes that produce DDL events cannot occur on the source cluster during the 
  migration. The following events cannot occur: 
  
  - create
  - drop
  - dropDatabase
  - createIndexes
  - dropIndexes
  - collModrefineCollectionShardKey
  - rename
  - reshardCollection

>   - shardCollection

Reviewer: [question]

Should we monospace these?

Author: ah yes we should — overlooked that. updating!

  This includes operations that may create new collections such as 
  :dbcommand:`mapReduce`, :pipeline:`$out`, and :pipeline:`$merge`. This also 
  includes collections created implicitly from inserts. Only writes that produce 
  CRUD events can occur during the migration.

  .. note:: 
   
     Writes that produce DDL events on source collections outside of the 
     :ref:`namespace filter <c2c-filtered-sync>` are allowed.

> - :ref:`GeoHaystack indexes <index-geohaystack-index>` are not supported.

Reviewer: [Question]

Do we need to mention this? These have been removed since 5.0 and we don't have any docs on them anymore.

I think if we're keeping this, we want `geoHaystack` to be in camel case with a lower-case `g`.

Reviewer 2: Yup I agree with Ali. Also I think we still have some docs on them: https://www.mongodb.com/docs/manual/core/indexes/index-types/geospatial/geohaystack/

Reviewer 2: yup that works!

- :ref:`/reverse <c2c-api-reverse>` endpoints are not supported. You can't 
  enable ``reversible`` in the :ref:`/start <c2c-api-start>` request.

> - You can't enable the ``enableUserWriteBlocking`` in the ``/start`` request. 

Reviewer: ```suggestion
- You can't enable ``enableUserWriteBlocking`` in the ``/start`` request. 
```

Reviewer: Or maybe "enable the `enableUserWriteBlocking` option". Whichever you prefer.

Author: fixed/updated

> - :ref:`SRV connection strings <connections-dns-seedlist>` for 4.4 source 

Reviewer: Making sure I'm understanding this correctly - this limitation only applies to 4.4 source clusters, not all pre-6.0 clusters?

Author: @ajayvijayakumar123 can you confirm on whether this is true?

Reviewer 2: yup just for 4.4 sources

Reviewer: Maybe we can reorder the phrases here slightly so it's clearer on a quick scan that this only applies to 4.4. Like:

"For 4.4 source clusters, [SRV connection strings](https://www.mongodb.com/docs/master/reference/connection-string/#std-label-connections-dns-seedlist) are not supported. You must use a [standard connection string.](https://www.mongodb.com/docs/master/reference/connection-string/#std-label-connections-standard-connection-string-format)"

I'd also maybe consider moving this to the end of the list since it's kind of in its own category of only applying to 4.4 clusters.

  clusters are not supported. You must use a :ref:`standard connection string
  <connections-standard-connection-string-format>`.

- You can't enable the ``createSupportingIndexes`` :ref:`sharding parameter 
  <c2c-api-start-sharding>` with a pre-6.0 cluster. 

> - Indexes with inconsistent specs or are missing on some shards are not 

Reviewer: ```suggestion
- Indexes with inconsistent specs or that are missing on some shards are not 
```

Author: Fixed

> - Writes that produce DDL events cannot occur on the source cluster during the 

Reviewer 2: should this PR be based against `mongodb:feature-older-version-support` feature branch instead?

Author: a h, thanks for catching that — fixed.

  migration. The following events cannot occur: 
  
  - ``create``
  - ``drop``
  - ``dropDatabase``
  - ``createIndexes``
  - ``dropIndexes``
  - ``collModrefineCollectionShardKey``
  - ``rename``
  - ``reshardCollection``

>   - ``shardCollection``

Reviewer: [Suggestion]

Maybe alphabetize?

Author: I'm inclined to leave this as is, or at least with few swaps, because I felt it'd be good to order it this way to users see the most familiar events at the top. If we swapped this to alphabetical, I though it could be a bit jarring for users to first see ``collModRefineCollectionShardKey``

Author: Or actually, @ajayvijayakumar123 just to double check is this supposed to be `collModrefineCollectionShardKey` or is this supposed to be `collMod` and `refineCollectionShardKey` separately?

Reviewer 2: yaa you're right lol. i made a lil typo 😅

  This includes operations that may create new collections such as 
  :dbcommand:`mapReduce`, :pipeline:`$out`, and :pipeline:`$merge`. This also 
  includes collections created implicitly from inserts. Only writes that produce 
  CRUD events can occur during the migration.

  .. note:: 
   
     Writes that produce DDL events on source collections outside of the 
     :ref:`namespace filter <c2c-filtered-sync>` are allowed.

- ``geoHaystack`` indexes are not supported.

> - :ref:`/reverse <c2c-api-reverse>` endpoints are not supported. You can't 

Reviewer 2: ```suggestion
- :ref:`/reverse <c2c-api-reverse>` endpoint is not supported. You can't 
```

> - :ref:`/reverse <c2c-api-reverse>` endpoints are not supported. You can't 

Reviewer 2: Are you planning to update the reverse limitation on the reverse page in a separate PR?

  enable the ``reversible`` option in the :ref:`/start <c2c-api-start>` request.

- You can't enable the ``enableUserWriteBlocking`` option in the ``/start`` 
  request. 

- You can't enable the ``createSupportingIndexes`` :ref:`sharding parameter 

>   <c2c-api-start-sharding>` with a pre-6.0 cluster. 

Reviewer 2: ```suggestion
  <c2c-api-start-sharding>`. 
```

since this whole section is for pre-6.0 migrations

> - Writes that produce DDL events cannot occur on the source cluster during the 

Reviewer 3: can we add a link to the definition of DDL?

  migration. The following events cannot occur: 
  
  - ``collMod``
  - ``create``
  - ``createIndexes``
  - ``drop``
  - ``dropDatabase``
  - ``dropIndexes``
  - ``refineCollectionShardKey``  
  - ``rename``
  - ``reshardCollection``
  - ``shardCollection``

  This includes operations that may create new collections such as 
  :dbcommand:`mapReduce`, :pipeline:`$out`, and :pipeline:`$merge`. This also 
  includes collections created implicitly from inserts. Only writes that produce 
  CRUD events can occur during the migration.

  .. note:: 
   
     Writes that produce DDL events on source collections outside of the 
     :ref:`namespace filter <c2c-filtered-sync>` are allowed.

- ``geoHaystack`` indexes are not supported.

- :ref:`/reverse <c2c-api-reverse>` endpoint is not supported. You can't 
  enable the ``reversible`` option in the :ref:`/start <c2c-api-start>` request.

> - You can't enable the ``enableUserWriteBlocking`` option in the ``/start`` 

Reviewer 3: can we add a note here that the user is responsible for preventing writes to the source or destination?

Author: @mvankeulen94 @ajayvijayakumar123 do we have a way of ensuring users don't write to their clusters _without_ the `enableUserWriteBlocking` option?

Reviewer 2: I think they just need to manually make sure

  request. 

- You can't enable the ``createSupportingIndexes`` :ref:`sharding parameter 

>   <c2c-api-start-sharding>`. 

Reviewer 3: @ajayvijayakumar123 is there a workaround we can advise for users? It's tricky to digest what this parameter does if the reader isn't already familiar with it

Reviewer 2: Yup. Let's say the user wants to shard on shard key `a:1`. Instead of enabling `createSupportingIndexes,` the user would need to create an index to support the shard key on the source.

- Indexes with inconsistent specs or that are missing on some shards are not 
  supported. To check for index inconsistencies, see 

>   :ref:`manage-indexes-find-inconsistent-indexes`.

Reviewer 3: @ajayvijayakumar123 is this a change from status quo mongosync GA? We have 'mongosync only syncs indexes that have consistent index specifications on all shards.' in the limitations today

Reviewer 2: Yup. For 6.0+ migrations, mongosync _ignores_ inconsistent indexes and just doesn't build them on the destination. For pre-6.0 migrations, mongosync will error on these indexes.

Author: I'll make it clear that it'll error on pre-6.0 migrations

Reviewer 3: thanks!
